gdnative-core/src/export/method.rs

```diff
@@ -282,18 +282,132 @@ impl<'a> Varargs<'a> {
     pub unsafe fn from_sys(num_args: libc::c_int, args: *mut *mut sys::godot_variant) -> Self {
         let args = std::slice::from_raw_parts(args, num_args as usize);
         let args = std::mem::transmute::<&[*mut sys::godot_variant], &[&Variant]>(args);
-        Self {
-            idx: 0,
-            iter: args.iter(),
-        }
+        Self { idx: 0, args }
     }
 }
 
 impl<'a> Iterator for Varargs<'a> {
     type Item = &'a Variant;
     #[inline]
     fn next(&mut self) -> Option<Self::Item> {
-        self.iter.next().copied()
+        if self.idx < self.args.len() {
+            let idx = self.idx;
+            self.idx += 1;
+            Some(self.args[idx])
+        } else {
+            None
+        }
```

Comment:
This part may be simplified by using `slice.get`?
```suggestion
        let result = self.slice.get(self.idx);
        self.idx += 1;
        result
```

Reply:
The test did not pass, so I made some fixes.
```suggestion
        let ret = self.args.get(self.idx);
        ret.map(|&v| {
            self.idx += 1;
            v
        })
```

Reply:
This is slightly confusing, as `map()` typically transforms the contents of the `Option` and has no side effects.

I'd rather do:
```rs
let ret = self.args.get(self.idx);
if ret.is_some() {
    self.idx += 1;
}
ret
```

In the future there might be [`Option::inspect()`](https://doc.rust-lang.org/std/option/enum.Option.html#method.inspect) for this purpose, but it's not yet stable.

Reply:
Fixed. Reflected in next push.